### PR TITLE
libmbp: Initial attempt at 'Optimus L Series' Boot UI support

### PR DESCRIPTION
**Pull request by Caio99BR**

### CODE REVIEW

File: libmbp/devices/lg.cpp [hunk 1]
```diff
@@ -91,58 +91,94 @@ void addLgOptimusGSeriesPhones(std::vector<Device *> *devices)
     devices->push_back(device);
 }
 
+void CommonLg7x27a()
+{
+    device->setSystemBlockDevs({ "/dev/block/mmcblk0p14",
```
Comment (chenxiaolong) [inline libmbp/devices/lg.cpp]: This won't compile. "device" is not passed to this function.

**Comment by Caio99BR** [comment]
Fixed!

File: libmbp/devices/lg.cpp [hunk 2]
```diff
@@ -91,58 +91,94 @@ void addLgOptimusGSeriesPhones(std::vector<Device *> *devices)
     devices->push_back(device);
 }
 
-void addLgOptimusLSeriesPhones(std::vector<Device *> *devices)
+static void CommonLgOptimusLSeries7x27a(std::vector<Device *> *devices)
```
Comment (chenxiaolong) [inline libmbp/devices/lg.cpp]: static void CommonLgOptimusLSeries7x27a(Device *device)

File: libmbp/devices/lg.cpp [hunk 3]
```diff
@@ -91,58 +91,94 @@ void addLgOptimusGSeriesPhones(std::vector<Device *> *devices)
...
-    device->setRecoveryBlockDevs({ "/dev/block/mmcblk0p17", "/dev/block/platform/msm_sdcc.3/by-num/p17" });
+    device->setId("m4");
+    device->setCodenames({
+        // Main Codename
+        "m4",
+        // Variant Names
+        "e610", "e612", "e617",
+        "E610", "E612", "E617" });
+    device->setName("LG Optimus L5");
+    CommonLgOptimusLSeries7x27a();
```
Comment (chenxiaolong) [inline libmbp/devices/lg.cpp]: CommonLgOptimusLSeries7x27a(device);

File: libmbp/devices/lg.cpp [hunk 4]
```diff
@@ -91,58 +91,94 @@ void addLgOptimusGSeriesPhones(std::vector<Device *> *devices)
...
-    device->setRecoveryBlockDevs({ "/dev/block/mmcblk0p17", "/dev/block/platform/msm_sdcc.3/by-num/p17" });
+    device->setId("u0");
+    device->setCodenames({
+        // Main Codename
+        "u0",
+        // Variant Names
+        "p700", "p705", "p708",
+        "P700", "P705", "P708" });
+    device->setName("LG Optimus L7");
+    CommonLgOptimusLSeries7x27a();
```
Comment (chenxiaolong) [inline libmbp/devices/lg.cpp]: CommonLgOptimusLSeries7x27a(device);

File: libmbp/devices/lg.cpp [hunk 5]
```diff
@@ -91,58 +91,94 @@ void addLgOptimusGSeriesPhones(std::vector<Device *> *devices)
...
-    device->setRecoveryBlockDevs({ "/dev/block/mmcblk0p17", "/dev/block/platform/msm_sdcc.3/by-num/p17" });
+    device->setId("v1");
+    device->setCodenames({
+        // Main Codename
+        "v1", "v1ds",
+        // Variant Names
+        "e410", "e411", "e415", "e420",
+        "E410", "E411", "E415", "E420" });
+    device->setName("LG Optimus L1 II");
+    CommonLgOptimusLSeries7x27a();
```
Comment (chenxiaolong) [inline libmbp/devices/lg.cpp]: CommonLgOptimusLSeries7x27a(device);

File: libmbp/devices/lg.cpp [hunk 6]
```diff
@@ -91,58 +91,94 @@ void addLgOptimusGSeriesPhones(std::vector<Device *> *devices)
...
+    device = new Device();
+    device->setId("vee3");
+    device->setCodenames({
+        // Main Codename
+        "vee3", "vee3ds",
+        // Variant Names
+        "e425", "e430", "e431", "e435",
+        "E425", "E430", "E431", "E435" });
+    device->setName("LG Optimus L3 II");
+    CommonLgOptimusLSeries7x27a();
```
Comment (chenxiaolong) [inline libmbp/devices/lg.cpp]: CommonLgOptimusLSeries7x27a(device);

File: libmbp/devices/lg.cpp [hunk 7]
```diff
@@ -91,58 +91,94 @@ void addLgOptimusGSeriesPhones(std::vector<Device *> *devices)
...
-    device->setBootBlockDevs({ "/dev/block/mmcblk0p9", "/dev/block/platform/msm_sdcc.3/by-num/p9" });
-    device->setRecoveryBlockDevs({ "/dev/block/mmcblk0p17", "/dev/block/platform/msm_sdcc.3/by-num/p17" });
+    device->setCodenames({
+        // Main Codename
+        "vee7", "vee7ds",
+        // Variant Names
+        "p710", "p712", "p713", "p714", "p715", "p716",
+        "P710", "P712", "P713", "P714", "P715", "P716" });
+    device->setName("LG Optimus L7 II");
+    CommonLgOptimusLSeries7x27a();
```
Comment (chenxiaolong) [inline libmbp/devices/lg.cpp]: CommonLgOptimusLSeries7x27a(device);

File: libmbp/devices/lg.cpp (hunk 2 above)
Comment (Caio99BR) [inline libmbp/devices/lg.cpp]: But why it?

Comment (chenxiaolong) [inline libmbp/devices/lg.cpp]: You want to pass the device variable you're modifying, not the entire list of devices.

**Comment by Caio99BR** [comment]
Fixed

File: libmbp/devices/asus.cpp [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2015  Andrew Gunnerson <andrewgunnerson@gmail.com>
+ * Copyright (C) 2016  Andrew Gunnerson <andrewgunnerson@gmail.com>
```
Comment (chenxiaolong) [inline libmbp/devices/asus.cpp]: These should all be `2015-2016`. Sorry to nitpick, but I can get in trouble (with my employer) if this is not done properly :)

Comment (Caio99BR) [inline libmbp/devices/asus.cpp]: Ok :P

**Comment by Caio99BR** [comment]
Done!

**Comment by Caio99BR** [comment]
Wait a moment, i'm fixing some devices names, i don't want to open another PR

**Comment by chenxiaolong** [comment]
No need to open new PRs. I can squash all your commits into 1 when merging.

**Comment by Caio99BR** [comment]
Ok but squash not is very good

**Comment by Caio99BR** [comment]
Also, it's time to check

**Comment by chenxiaolong** [comment]
@chenxiaolong-bot: test this please

**Comment by Caio99BR** [comment]
Idk if is my internet, but i need to use VPN to download the file not fully corrurpted (file still corrupted partially, and UI weird).
But Ok, for me it works :)

**Comment by chenxiaolong** [comment]
Merged, thank you! In the future, I will create either a `clang-format` or `astyle` file, which will automatically format the files properly (including 80-char wrap).

**Comment by Caio99BR** [comment]
Look good, also thank you for this BootUI changes :)

2 things, you need to update translations to Transifex and add translations to BootUi too.
Ah don't forget to fix /cache/multiboot when installing BootUI, I think is only this notes.

Thanks for all!

**Comment by chenxiaolong** [comment]
> 2 things, you need to update translations to Transifex and add translations to BootUi too.

Will do. I'll try and see if Boot UI translations will work on Transifex. It's not exactly the same format as Android.

> Ah don't forget to fix /cache/multiboot when installing BootUI, I think is only this notes.

This is fixed now :) https://github.com/chenxiaolong/DualBootPatcher/commit/4538fe9222f036c39146b130a99a1402fb632a29

**Comment by Caio99BR** [comment]
> Will do. I'll try and see if Boot UI translations will work on Transifex. It's not exactly the same format as Android.

Ok

> This is fixed now :) 4538fe9

It create new error #116 ;)

**Pull request merged**